pkg/client/unversioned/request.go

```diff
@@ -115,17 +115,22 @@ type Request struct {
 }
 
 // NewRequest creates a new request helper object for accessing runtime.Objects on a server.
-func NewRequest(client HTTPClient, verb string, baseURL *url.URL, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
+func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPath string, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
 	if backoff == nil {
 		glog.V(2).Infof("Not implementing request backoff strategy.")
 		backoff = &NoBackoff{}
 	}
 	metrics.Register()
+
+	pathPrefix := "/"
+	if baseURL != nil {
+		pathPrefix = path.Join(pathPrefix, baseURL.Path)
+	}
 	return &Request{
 		client:       client,
 		verb:         verb,
 		baseURL:      baseURL,
-		path:         baseURL.Path,
+		path:         path.Join(pathPrefix, versionedAPIPath),
```

Comment:
I guess I don't get why you store path with the prefix. Why not just join baseURL and path at the very end when you're about to make the request, and keep them separate until then?

(sorry for slow responses; I have limited availability last week and this week)